<a href="https://colab.research.google.com/github/NikitaMarnykh/labor_market_analysis/blob/main/%D0%9C%D0%B0%D1%80%D0%BD%D1%8B%D1%85_%D0%9D_%D0%92_%D0%B8_%D0%9E%D0%B1%D0%BE%D1%80%D0%BE%D1%82%D0%BE%D0%B2_%D0%9C_%D0%92_%D0%B0%D0%BD%D0%B0%D0%BB%D0%B8%D0%B7_%D1%80%D1%8B%D0%BD%D0%BA%D0%B0_%D1%82%D1%80%D1%83%D0%B4%D0%B0_advanced_plus.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 1. Постановка задачи и описание датасета

Перед Департаментом труда и занятости населения Томской области стоит задача
– в течение полугода разработать эффективные стратегии управления
персоналом, образования и подготовки кадров в целях увеличения уровня
занятости в регионе на 2 процентных пункта. Для этого специалистам необходимо
лучше понять особенности функционирования рынка труда, а именно: в какие
города области и какие специалисты требуются, какими ключевыми навыками они
должны обладать, какие условия труда предлагают работодатели, есть ли
возможность трудоустроиться соискателям с инвалидностью.
Специалистами Роснавыка были представлены датасеты с выгруженной
информацией по региону.

**Задачи:**

  - Проанализируйте ключевые навыки в самой популярной сфере/сферах в регионе и выявить значимые взаимосвязи между признаками (на какие навыки и в каких городах наибольший спрос, есть ли взаимосвязь между навыками и требованиями к соискателям, как тот или иной навык влияет на условия труда) (датасет hh_hard.csv)

  - построить регрессионную модель для предсказания уровня заработной платы.

  - Представьте аналитический отчет содержащий выводы и рекомендации для Заказчика, которые помогут ему для решения его бизнес-задач.

**Исходные данные:**

  - идентификатор вакансии (id)

  - наименование вакансии (name)

  - профессиональная роль (professional roles)

  - регион (area), в т.ч. страна (country), регион (region), округ (district), город(city)

  - заработная плата (salary), в т.ч. минимальная заработная плата (salary_from),
максимальная заработная плата (salary_to)

  - опыт работы (experience): 0 – без опыта, 1-3 – от 1 до 3 лет, 3-6 – от 3 до 6 лет, 6- - от 6 лет

  - график работы (schedule): full_day – полный день, remote – удаленная работа,
  flexible – гибкий график, shift – сменный график, labor - вахтовый метод

  - тип занятости (employment): full – полная занятость, part – частичная
занятость, project – проектная работа, probation - стажировка

  - наименование работодателя (employer)

  - доступна ли вакансия для соискателей с инвалидностью (accept_handicapped): True – доступна, False - недоступна

  - ключевые навыки (key_skills)

**Датасет в формате csv по ссылке:**

  - https://disk.yandex.ru/d/OYP_n5bYz_ExzA


## 2. Установка и подключение зависимостей

Устанавливаем все необходимые зависимости

In [ ]:
!pip install numpy
!pip install matplotlib
!pip install pandas
!pip install scipy
!pip install seaborn
!pip install statsmodels
!pip install scikit-learn

Подключаем все необходимые зависимости

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats
import seaborn as sns

Подгружаем репозиторий для работы с ним

In [ ]:
!git clone https://github.com/NikitaMarnykh/labor_market_analysis

Cloning into 'labor_market_analysis'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 14 (delta 2), reused 9 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (14/14), 1.40 MiB | 4.36 MiB/s, done.
Resolving deltas: 100% (2/2), done.


Перейдём в установленный репозиторий

In [ ]:
%cd labor_market_analysis/

/content/labor_market_analysis


## 3. Первичный обзор датасета. Оценка размеров датасета, наличия дубликатов и пропусков

Сохраняем датасет в переменную

In [ ]:
dataset = pd.read_csv('hh_hard.csv',
                      na_values=['NA', 'N/A', 'null', 'missing', '-', '?', '...'],
                      keep_default_na=True,
                      na_filter=True)

Проведём обзор содержимого таблицы

In [ ]:
dataset.head(15)

id                                               name  \
0   72121675  Инженер-конструктор (в г. Электрогорск, Москов...   
1   72121730                Специалист по доставке (Подработка)   
2   72122239   Удаленный оператор ПК / сотрудник на ввод данных   
3   72123219                                Повар линии раздачи   
4   72123390             Повар в магазин SPAR (ул.Киевская 139)   
5   72124633                                     Дефектоскопист   
6   72124864                                 Водитель-кладовщик   
7   72125286                                      Электромонтер   
8   72125786                               Менеджер по продажам   
9   72125979                               Менеджер по продажам   
10  72127032       Ведущий специалист обработки корреспонденции   
11  72127862                            Ведущий инженер-сметчик   
12  72127872                            Ведущий инженер-сметчик   
13  72127856                            Ведущий инженер-сметчик   
14  72127847                            Ведущий инженер-сметчик   

                                   professional_roles experience  schedule  \
0          Инженер-конструктор, инженер-проектировщик        1-3  full_day   
1                                              Курьер          0     shift   
2   Менеджер по продажам, менеджер по работе с кли...          0    remote   
3                           Официант, бармен, бариста          0     shift   
4                             Повар, пекарь, кондитер          0  full_day   
5                                            Технолог        1-3  full_day   
6                                            Водитель          0  full_day   
7                                    Электромонтажник        3-6     labor   
8   Менеджер по продажам, менеджер по работе с кли...        1-3  full_day   
9                               Агент по недвижимости          0  full_day   
10                      Делопроизводитель, архивариус          0  full_day   
11                       Инженер ПТО, инженер-сметчик        1-3  full_day   
12                       Инженер ПТО, инженер-сметчик        1-3  full_day   
13                       Инженер ПТО, инженер-сметчик        1-3  full_day   
14                       Инженер ПТО, инженер-сметчик        1-3  full_day   

   employment                  employer  accept_handicapped  \
0        full       Стройкомплектсервис               False   
1        part                    Квазар               False   
2        part              Staff-Linear                True   
3        full                 AYS GROUP               False   
4        full     ЛАМА, Группа компаний               False   
5        full             Деталь Проект               False   
6        full         Автоцентр АВТОГИК               False   
7        full               Эко-Системы               False   
8        full  Мототека, филиал г.Томск               False   
9        full                   ДомПлюс               False   
10       full                  МТС Банк               False   
11       full     ОГКУ Облстройзаказчик               False   
12       full     ОГКУ Облстройзаказчик               False   
13       full     ОГКУ Облстройзаказчик               False   
14       full     ОГКУ Облстройзаказчик               False   

                                           key_skills  \
0   ['ANSYS', 'AutoCAD', 'Autodesk Inventor', 'Sol...   
1                                  ['Грамотная речь']   
2             ['Грамотная речь', 'Базовые знания ПК']   
3   ['Грамотная речь', 'Грамотная речь', 'Работа в...   
4   ['Коммуникативная грамотность', 'Обучаемость и...   
5   ['Ведение документооборота', 'Базовые знания П...   
6   ['Базовые знания ПК', 'Коммуникативная грамотн...   
7   ['Монтаж металлоконструкций', 'Выполнение стро...   
8   ['Продажи', 'Коммуникативная грамотность', 'Гр...   
9   ['Продажи', 'Продажи', 'Коммуникативная грамот...   
10  ['Microsoft Outlook', 'Microsoft PowerPoint', ..

Просмотрим информацию о датасете

In [ ]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20178 entries, 0 to 20177
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   id                  20178 non-null  int64 
 1   name                20178 non-null  object
 2   professional_roles  20178 non-null  object
 3   experience          20178 non-null  object
 4   schedule            20178 non-null  object
 5   employment          20178 non-null  object
 6   employer            20178 non-null  object
 7   accept_handicapped  20178 non-null  bool  
 8   key_skills          20178 non-null  object
 9   area                20178 non-null  object
 10  salary              20178 non-null  object
dtypes: bool(1), int64(1), object(9)
memory usage: 1.6+ MB


Информация, содержащаяся в датасете:
  
  - идентификатор вакансии (id)

  - наименование вакансии (name)

  - профессиональная роль (professional roles)

  - регион (area), в т.ч. страна (country), регион (region), округ (district), город(city)

  - заработная плата (salary), в т.ч. минимальная заработная плат (salary_from), максимальная заработная плата (salary_to)

  - опыт работы (experience)

  - график работы (schedule)

  - тип занятости (employment)

  - наименование работодателя (employer)

  - доступна ли вакансия для соискателей с инвалидностью (accept_handicapped)
  
  - ключевые навыки (key_skills)

Оценим размеры датасета

In [ ]:
dataset.shape

(20178, 11)

Оценим наличие полных дубликатов в датасете

In [ ]:
dataset.duplicated(keep=False).sum()

np.int64(356)

Оценим наличие пропущенных значений в датасете

In [ ]:
dataset.isnull().sum()

id                    0
name                  0
professional_roles    0
experience            0
schedule              0
employment            0
employer              0
accept_handicapped    0
key_skills            0
area                  0
salary                0
dtype: int64

Выводы по предварительному осмотру:

* **О размерах датасета**:

  20178 строк (наблюдений), 11 столбцов (признаков)

* **О типах данных**:

  - Количественные признаки:

    * идентификатор вакансии (id)

    * заработная плата (salary)

  - Категориальные признаки:

    * наименование вакансии (name)

    * профессиональная роль (professional roles)

    * регион (area)

    * опыт работы (experience)

    * график работы (schedule)

    * тип занятости (employment)

    * наименование работодателя (employer)

    * доступна ли вакансия для соискателей с инвалидностью (accept_handicapped)

    * ключевые навыки (key_skills)


* **О дубликатах**:

  - Присутствуют полные дубликаты


* **О пропущенных значениях:**

  - Пропущенных значений нет ни в одном из признаков

  - Возможно на этапе обработки признаков появятся нулевые значения там, где их быть не должно